runtimes/pythonrt/local_runner.go

```diff
@@ -145,6 +144,32 @@ func (r *LocalPython) Start(pyExe string, tarData []byte, env map[string]string,
 }
 
 func (r *LocalPython) Health() error {
+	if _, err := os.FindProcess(r.proc.Pid); err != nil {
+		return fmt.Errorf("runner heath: no runner proc found. %w ", err)
+	}
+
+	var status syscall.WaitStatus
+	if pid, err := syscall.Wait4(r.proc.Pid, &status, syscall.WNOHANG, nil); err == nil && pid == r.proc.Pid {
+		if status.Signaled() {
+			sig := status.Signal()
+			switch sig {
+			case syscall.SIGKILL, syscall.SIGTERM, syscall.SIGABRT, syscall.SIGSEGV, syscall.SIGBUS:
+				err = fmt.Errorf("runner health: proc terminated by signal %v", sig)
+			default:
+				// maybe process communicates with itself? do nothing meanwhile
+				return nil
+			}
+			if err == nil {
+				err = fmt.Errorf("runner health: unexpected proc state")
+			}
+
+			r.Close() // FIXME: not ideal :( should be called by the caller?
```

Comment:
I don't like this. Health should not cleanup. We can wait & hang in a goroutine and then do cleanup.

Reply:
you are totally right. I marked it with FIXME to discuss with you/Efi and get feedbacks how to implement it better. Bacause `close()` meanwhile not invoked on process stop/ ctx invalidation.

Reply:
removed meanwhile. Will handle this in different PR - see ENG-1675